dist/s_version

```diff
@@ -18,42 +19,66 @@ while :
 	esac
 done
 
-# If the version hasn't changed and we're not forcing the issue, we're done.
-if test "$force" = no -a \
-    -f $m4dir/version.m4 -a \
-    -f $m4dir/version-set.m4 ; then
-	eval `grep '^VERSION_[A-Z]*=' $m4dir/version-set.m4`
-	if test x${WIREDTIGER_VERSION_MAJOR} = x${VERSION_MAJOR} -a \
-		x${WIREDTIGER_VERSION_MINOR} = x${VERSION_MINOR} -a \
-		x${WIREDTIGER_VERSION_PATCH} = x${VERSION_PATCH} ; then
-		exit 0
+GenAutoconfVersion(){
+	# If the version hasn't changed and we're not forcing the issue, we're done.
+	if test "$force" = no -a \
+		-f $m4dir/version.m4 -a \
+		-f $m4dir/version-set.m4 ; then
+		eval `grep '^VERSION_[A-Z]*=' $m4dir/version-set.m4`
+		if test x${WIREDTIGER_VERSION_MAJOR} = x${VERSION_MAJOR} -a \
+			x${WIREDTIGER_VERSION_MINOR} = x${VERSION_MINOR} -a \
+			x${WIREDTIGER_VERSION_PATCH} = x${VERSION_PATCH} ; then
+			return
+		fi
 	fi
-fi
-
-dotted_version=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}.${WIREDTIGER_VERSION_PATCH}
-echo "Building $m4dir/version.m4"
-cat <<EOF > $m4dir/version.m4
-dnl WiredTiger product version for AC_INIT.  Maintained by dist/s_version
-${dotted_version}
-EOF
-
-echo "Building $m4dir/version-set.m4"
-cat <<EOF > $m4dir/version-set.m4
-dnl build by dist/s_version
-
-VERSION_MAJOR=${WIREDTIGER_VERSION_MAJOR}
-VERSION_MINOR=${WIREDTIGER_VERSION_MINOR}
-VERSION_PATCH=${WIREDTIGER_VERSION_PATCH}
-VERSION_STRING='"${WIREDTIGER_VERSION_STRING}"'
-
-AC_SUBST(VERSION_MAJOR)
-AC_SUBST(VERSION_MINOR)
-AC_SUBST(VERSION_PATCH)
-AC_SUBST(VERSION_STRING)
-
-VERSION_NOPATCH=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}
-AC_SUBST(VERSION_NOPATCH)
-EOF
+	dotted_version=${WIREDTIGER_VERSION_MAJOR}.${WIREDTIGER_VERSION_MINOR}.${WIREDTIGER_VERSION_PATCH}
+	echo "Building $m4dir/version.m4"
+	cat > $m4dir/version.m4 <<-END
+	dnl WiredTiger product version for AC_INIT.  Maintained by dist/s_version
+	${dotted_version}
+	END
```

Comment:
This seems to have changed from END to EOF. My scripting language is quite newbie. What is the difference?

Reply:
Just jumping in, the delimeter for a heredoc (multi-line bash string) can be anything `END`, `EOF`, `MYAWESOMEDELIMETER`. They just need to match (the beginning and end). Hence `END` & `EOF` are functionally equivalent. Conventionally speaking I usually see `END` (flagging the beginning and _end_ of a string).

Reply:
Ahh i see! Thanks so much for the explanation!!